analysis/study_definition.py

```diff
@@ -570,29 +572,125 @@
             "incidence": 0.95,
         },
     ),
+    # Fetch the comparator (<, >=, = etc) associated with a numeric value.
+    # Where a lab result is returned as e.g. <9.5 the numeric_value component
+    # will contain only the value 9.5 and you will need to use this function
+    # to fetch the comparator into a separate column.
+    # https://docs.opensafely.org/study-def-variables/#cohortextractor.patients.comparator_from
+    egfr_comparator=patients.comparator_from(
+        "egfr",
+        return_expectations={
+            "rate": "universal",
+            "category": {
+                "ratios": {  # ~, =, >= , > , < , <=
+                    None: 0.10,
+                    "~": 0.05,
+                    "=": 0.65,
+                    ">=": 0.05,
+                    ">": 0.05,
+                    "<": 0.05,
+                    "<=": 0.05,
+                }
+            },
+            "incidence": 0.80,
+        },
+    ),
+    egfr_category=patients.categorised_as(
+        {
+            "0": "DEFAULT",
+            "1": """
+                egfr_flag AND
+                    (egfr>=60 AND NOT
+                        (egfr_comparator = '<' AND NOT
+                        egfr_comparator = '<=' AND NOT
+                        egfr_comparator = '~'))
+            """,
+            "2": """
+                egfr_flag AND
+                    ((egfr>=45 AND NOT
+                        (egfr_comparator = '<' AND NOT
+                        egfr_comparator = '<=' AND NOT
+                        egfr_comparator = '~')) AND
+                            (egfr<60 AND NOT
+                                (egfr_comparator = '>' AND NOT
+                                egfr_comparator = '>=' AND NOT
+                                egfr_comparator = '~' AND NOT
+                                egfr_comparator = '=')))
+
```

Comment:
Could this be made simpler by saying `(egfr >= 45) AND (egrf<60) AND (egfr_comparator= '=')`?

Reply:
I might be misunderstanding this whole comparator thing. But suppose the value '>45' is returned, that should be included right? But it would not when I restrict to ```egfr_comparator = 1```.
Oh.... AHA! I understand why it actually shouldn't be included: if the value is '>45', we're not sure it's actually <60, so it shouldn't be included! So YES, it could be made simpler! Thx.